warehouse/postgres/postgresQueryBuilder.go

```diff
@@ -0,0 +1,63 @@
+package postgres
+
+import (
+	"database/sql"
+	"fmt"
+	"strings"
+)
+
+type QueryExecution struct {
+	txn                 *sql.Tx
+	db                  *sql.DB
+	query               string
+	enableWithQueryPlan bool
+}
+
+// handleQueryExecution
+// Print execution plan if enableWithQueryPlan is set to true else return result set.
+// Currently, these statements are supported by EXPLAIN ANALYZE
+// Any SELECT, INSERT, UPDATE, DELETE, VALUES, EXECUTE, DECLARE, CREATE TABLE AS, or CREATE MATERIALIZED VIEW AS statement, whose execution plan you wish to see.
+func handleQueryExecution(e *QueryExecution) (result sql.Result, err error) {
+	sqlStatement := e.query
+	if e.enableWithQueryPlan {
+		sqlStatement = "EXPLAIN ANALYZE " + e.query
+	}
+
+	if e.enableWithQueryPlan {
+		var rows *sql.Rows
+		if e.txn != nil {
+			rows, err = e.txn.Query(sqlStatement)
+		} else if e.db != nil {
+			rows, err = e.db.Query(sqlStatement)
+		} else {
+			err = fmt.Errorf("[WH][POSTGRES] Not able to handle query execution for statement: %s", sqlStatement)
+			return
+		}
+		if err != nil {
+			err = fmt.Errorf("[WH][POSTGRES] error occurred while handling transaction for query: %s with err: %w", sqlStatement, err)
+			return
+		}
+		defer rows.Close()
+
+		var response []string
+		for rows.Next() {
+			var s string
+			if err = rows.Scan(&s); err != nil {
+				err = fmt.Errorf("[WH][POSTGRES] Error occurred while processing destination revisionID query %+v with err: %w", e, err)
+				return
+			}
+			response = append(response, s)
+		}
+		pkgLogger.Infof("[WH][POSTGRES] Execution Query plan for statement: %s is %s", sqlStatement, strings.Join(response, "\n"))
```

Comment:
Is it okay to have `\n` in logs, would it make it hard to capture in Loki?

Reply:
Although logs in loki looks good. But it's coming in different line. It's better we should output it in a single line. 
<img width="1710" alt="Screenshot 2022-06-24 at 5 13 57 PM" src="https://user-images.githubusercontent.com/77991816/175527769-cbf5f3b7-da66-4e16-9eaf-96430f60b5cf.png">

Reply:
Using space separated.

Reply:
Using Raw string literal instead of \n now.

Reply:
But still this will comes as the corresponding output only. Explored other options like YAML, XML, JSON but this is the best options we have.

Reply:
👍